lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service.rb

```diff
@@ -201,14 +201,20 @@ def datastore
     def ldap
     end
 
-    def execute(method_name, *args)
-      User.with_user(@workspace.ae_user) { execute_with_user(method_name, *args) }
+    def execute(method_name, *args, **kwargs, &block)
+      User.with_user(@workspace.ae_user) { execute_with_user(method_name, *args, **kwargs, &block) }
     end
 
-    def execute_with_user(method_name, *args)
+    def execute_with_user(method_name, *args, **kwargs, &block)
       # Since each request from DRb client could run in a separate thread
       # We have to set the current_user in every thread.
-      MiqAeServiceMethods.send(method_name, *args)
+
+      # For ruby 2.6-3.0+ support, we grab any kwargs and append as a hash
+      # at the end of the args, as all MiqAeServiceMethods now don't accept
+      # kwargs.  When we get to ruby 3, we can remove this and convert all to
+      # kwargs.
+      args << kwargs unless kwargs.blank?
+      MiqAeServiceMethods.send(method_name, *args, &block)
```

Comment:
This is really the biggest change.  We capture the kwargs from ruby 2.7/3 and pass them along as a hash to the end of the args.  Then, we can keep all of the existing methods as accepting options hashes except for the one that was accepting kwargs, which I change below accept hashes instead.

There isn't another way to handle the situation in which existing automate code is calling execute with `:a =>` at that will be kwargs on ruby 3 even if we expect hashes.

Once we're on ruby 3+ only, we can use `**kwargs` and always pass along kwargs and all the methods can be made to accept hashes as kwargs.